rainier-core/src/main/scala/com/stripe/rainier/sampler/LeapFrog.scala

```diff
@@ -122,47 +119,69 @@ private[sampler] case class LeapFrog(nVars: Int,
 }
 
 private[sampler] object LeapFrog {
-  def apply(context: Context): LeapFrog = {
-    val qs = context.variables
-    val ps = qs.map { _ =>
-      new Variable
+  def apply(densityFun: DensityFunction): LeapFrog = {
+
+    def split(inputs: Array[Double])
```

Comment:
we need to find a way to do this with zero allocation - previously when we had similar split() logic to this, we were getting hosed with GC when doing enough iterations of large enough modes

Reply:
Yeah, I figured we'd have to fix this. Okay, I'll work on this.